tools/importer/importer.py

```diff
@@ -0,0 +1,254 @@
+            mkdir(os.path.dirname(abs_dst_file))
+            copy_file(abs_src_file, abs_dst_file)
+
+def run_cmd_with_output(command, exit_on_failure=False):
+    """ Passes a command to the system and returns a True/False result once the
+        command has been executed, indicating success/failure. If the command was
+        successful then the output from the command is returned to the caller.
+        Commands are passed as a list of tokens.
+        E.g. The command 'git remote -v' would be passed in as ['git', 'remote', '-v']
+
+    Args:
+    command - system command as a list of tokens
+    exit_on_failure - If True exit the program on failure (default = False)
+
+    Returns:
+    result - True/False indicating the success/failure of the command
+    output - The output of the command if it was successful, else empty string
+    """
+    rel_log.debug('[Exec] %s', ' '.join(command))
+    returncode = 0
+    output = ""
+    try:
+        output = subprocess.check_output(command, shell=True)
+    except subprocess.CalledProcessError as e:
+        returncode = e.returncode
+
+        if exit_on_failure:
+            rel_log.error("The command %s failed with return code: %s",
+                        (' '.join(command)), returncode)
+            sys.exit(1)
+    return returncode, output
+
+def get_curr_sha(repo_path):
+    """ Gets the latest SHA for the specified repo
+    Args:
+    repo_path - path to the repository
+
+    Returns:
+    sha - last commit SHA
+    """
+    cwd = os.getcwd()
+    os.chdir(abspath(repo_path))
+
+    cmd = "git log --pretty=format:%h -n 1"
+    _, sha = run_cmd_with_output(cmd, exit_on_failure=True)
+
+    os.chdir(cwd)
+    return sha
+
+def check_branch(name):
+    """ Check if branch already exists in mbed-os local repository.
+    It will not verify if branch is present in remote repository.
+    Args:
+    name - branch name
+    Returns:
+    True - If branch is already present
+    """
+
+    cmd = "git branch"
+    _, output = run_cmd_with_output(cmd, exit_on_failure=False)
+    if name in output:
+        return True
+    return False
+
+def branch_checkout(name):
+    """
+    Checkout the required branch
+    Args:
+    name - branch name
+    """
+    cmd = "git checkout " + name
+    run_cmd_with_output(cmd, exit_on_failure=False)
+
+def get_last_cherry_pick_sha(branch):
+    """
+    SHA of last cherry pick commit is returned. SHA should be added to all
+    cherry-pick commits with -x option.
+
+    Args:
+    branch - Hash to be verified.
+    Returns - SHA if found, else None
+    """
+    cmd = "git checkout " + branch
+    run_cmd_with_output(cmd, exit_on_failure=False)
+
+    SHA = None
+    get_commit = "git log -n 1"
+    _, output = run_cmd_with_output(get_commit, exit_on_failure=True)
+    lines = output.split('\n')
+    for line in lines:
+        if 'cherry picked from' in line:
+            SHA = line.split(' ')[-1]
+            SHA = SHA[:-1]
```

Comment:
Could this be `return SHA[:-1]`? I think there is only ever 1 or 0 lines that match this filter, so taking the last one is the same as taking the first one.